frameworks/hdfs/tests/config.py

```diff
@@ -74,7 +75,23 @@ def run_hdfs_command(svc_name, command):
     return rc, output
 
 
-def check_healthy(count=DEFAULT_TASK_COUNT):
-    sdk_plan.wait_for_completed_deployment(PACKAGE_NAME, timeout_seconds=25 * 60)
-    sdk_plan.wait_for_completed_recovery(PACKAGE_NAME, timeout_seconds=25 * 60)
-    sdk_tasks.check_running(PACKAGE_NAME, count)
+def check_healthy(service_name=PACKAGE_NAME, count=DEFAULT_TASK_COUNT, recovery_expected=False):
+    sdk_plan.wait_for_completed_deployment(service_name, timeout_seconds=25 * 60)
+    if recovery_expected:
+        # TODO(elezar): See INFINITY-2109 where we need to better handle recovery health checks
+        sdk_plan.wait_for_kicked_off_recovery(service_name, timeout_seconds=25 * 60)
+    sdk_plan.wait_for_completed_recovery(service_name, timeout_seconds=25 * 60)
+    sdk_tasks.check_running(service_name, count)
+
+
+def expect_recovery(service_name=PACKAGE_NAME):
+    # TODO(elezar, nima) check_healthy also check for complete deployment, and this should not
+    # affect the state of recovery.
+    check_healthy(service_name=service_name, count=DEFAULT_TASK_COUNT, recovery_expected=True)
+
+
+def get_pod_type_instances(pod_type_prefix, service_name=FOLDERED_SERVICE_NAME):
+    pod_types = sdk_cmd.convert_output_to_list(
```

Comment:
I've tested this locally and `ast.literal_eval()` takes care of that for us. It turns a list of elements represented as a newline-separated string into a list type. We then do the strip on each element in the list compression in `convert_output_to_list` in case there's extra whitespace (though that's just precautionary).